content/en/blog/_posts/2024-08-14-Kubernetes-v1-31-Release.md

```diff
@@ -0,0 +1,302 @@
+Maintance mode means that no new features will be added to cgroup v1 support. 
+Critical security fixes will still be provided, however, bug-fixing is now best-effort, meaning major bugs may be fixed if feasible, but some issues might remain unresolved.
+
+Its recommended that you start switching to use cgroup v2 as soon as possible. For more details on the reasoning behind placing cgroup v1 in maintaince mode and starting the migration process see the [Moving cgroup v1 Support into Maintenance Mode in Kubernetes 1.31](/blog/kubernetes-1-31-moving-cgroup-v1-support-maintenance-mode) article. 
+
+Please report any problems encounter by filing an [issue](https://github.com/kubernetes/kubernetes).
+
+This work was done as part of [KEP #4569](https://github.com/kubernetes/enhancements/issues/4569) by [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node).
+
+## Release notes and upgrade actions required
+
+Check out the full details of the Kubernetes 1.31 release in our [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.31.md).
+
+### Scheduler now uses QueueingHint when `SchedulerQueueingHints` is enabled
+Added support to the scheduler to start using a QueueingHint registered for Pod/Updated events,
+to determine whether updates to  previously unschedulable Pods have made them schedulable.
+The new support is active when the feature gate `SchedulerQueueingHints` is enabled.
+
+Previously, when unschedulable Pods were updated, the scheduler always put Pods back to into a queue
+(`activeQ` / `backoffQ`). However not all updates to Pods make Pods schedulable, especially considering
+many scheduling constraints nowadays are immutable. Under the new behaviour, once unschedulable Pods
+are updated, the scheduling queue checks with QueueingHint(s) whether the update may make the
+pod(s) schedulable, and requeues them to `activeQ` or `backoffQ` only when at least one
+QueueingHint returns `Queue`.
+
+**Action required for custom scheduler plugin developers**: 
+Plugins have to implement a QueueingHint for Pod/Update event if the rejection from them could be resolved by updating unscheduled Pods themselves. Example: suppose you develop a custom plugin that denies Pods that have a `schedulable=false` label. Given Pods with a `schedulable=false` label will be schedulable if the `schedulable=false` label is removed, this plugin would implement QueueingHint for Pod/Update event that returns Queue when such label changes are made in unscheduled Pods. [#122234](https://github.com/kubernetes/kubernetes/pull/122234)
+
+### Removal of kubelet --keep-terminated-pod-volumes command line flag
+The kubelet flag `--keep-terminated-pod-volumes`, which was deprecated in 2017, was removed as part of the v1.31 release.
+
+You can find more details in the pull request [#122082](https://github.com/kubernetes/kubernetes/pull/122082).
+
+
+## Availability
+
```

Comment:
BTW, we could also mention that https://kubernetes.io/releases/download/ exists.

Reply:
Here's the fixed version

```
Kubernetes 1.31 is available for download on [GitHub](https://github.com/kubernetes/kubernetes/releases/tag/v1.31.0) or on the [Kubernetes download page](https://kubernetes.io/releases/download/). 

To get started with Kubernetes, check out these [interactive tutorials](/docs/tutorials/) or run local Kubernetes clusters using [minikube](https://minikube.sigs.k8s.io/). You can also easily install 1.31 using [kubeadm](/docs/setup/independent/create-cluster-kubeadm/). 
```

Reply:
updated.